modules/phenyl-api-explorer/src/PhenylApiExplorer.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-env node */
+// @flow
+import fs from 'fs'
+import path from 'path'
+import ejs from 'ejs'
+import type {
+  FunctionalGroup,
+  EncodedHttpResponse,
+} from 'phenyl-interfaces'
+
+export type ExplorerParams = {|
+  path: string,
+|}
+
+export default class PhenylApiExplorer {
+  functionalGroup: FunctionalGroup;
+  path: string;
+  handler: any;
+  html: string;
+
+  constructor (functionalGroup: FunctionalGroup, params: ExplorerParams) {
+    this.functionalGroup = functionalGroup
+    this.path = params.path
+    this.handler = this.handler.bind(this)
+    this.html = this.getClientHTML()
+  }
+
+  shallowMap (obj: Object, fn: (any, string, Object) => any): Object {
+    const ret = {}
+    for (let p in obj) {
+      ret[p] = fn(obj[p], p, obj)
+    }
+    return ret
+  }
+
+  getClientHTML (): string {
+    const templatePath = path.join(__dirname, 'client', 'build', 'index.html')
+    const template = fs.readFileSync(templatePath, 'utf8')
+    const data = {
+      functionalGroup: {
+        users: this.shallowMap(this.functionalGroup.users, ({ accountPropName, passwordPropName }) => {
```

Comment:
This depends on standard definition. We should discuss how to pass extra data to the renderer.

Reply:
Let's throw an error indicating that `passwordPropName` and `accountPropName` are required properties.